File: Python/classe-de-heroi.py
class Heroi:
    
    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        if self.tipo.lower() == 'mago':
            ataque = 'magia'
        elif self.tipo.lower() == 'guerreiro':
            ataque = 'espada'
        elif self.tipo.lower() == 'monge':
            ataque = 'artes marciais'
        elif self.tipo.lower() == 'ninja':
            ataque = 'shuriken'
        else:
            ataque = 'ataque desconhecido'

        print(f'O {self.tipo} atacou usando {ataque}')

herois = [
    Heroi('Gandalf', 300, 'Mago'),
    Heroi('Aragorn', 87, 'Guerreiro'),
    Heroi('Kung Fu Panda', 20, 'Monge'),
    Heroi('Naruto', 16, 'Ninja')
]

for heroi in herois:
    heroi.atacar()

File: Java/classe-de-heroi.java
class Heroi {

    public String nome;
    public int idade;
    public String tipo;

    public Heroi(String nome, int idade, String tipo) {
        this.nome = nome;
        this.idade = idade;
        this.tipo = tipo;
    }

    public void atacar() {
        String ataque;
        switch (tipo.toLowerCase()) {
            case "mago":
                ataque = "magia";
                break;
            case "guerreiro":
                ataque = "espada";
                break;
            case "monge":
                ataque = "artes marciais";
                break;
            case "ninja":
                ataque = "shuriken";
                break;
            default:
                ataque = "ataque desconhecido";
        }
        System.out.println("O " + tipo + " atacou usando " + ataque);
    }

    public static void main(String[] args) {
        Heroi[] herois = {
            new Heroi("Gandalf", 300, "Mago"),
            new Heroi("Aragorn", 87, "Guerreiro"),
            new Heroi("Kung Fu Panda", 20, "Monge"),
            new Heroi("Naruto", 16, "Ninja")
        };

        for (Heroi heroi : herois) {
            heroi.atacar();
        }
    }
}